src/quacc/atoms/skzcam.py

```diff
@@ -6,35 +6,173 @@
 import numpy as np
 from ase.atoms import Atoms
 from ase.data import atomic_numbers
-from ase.io import write
+from ase.io import read, write
 from ase.units import Bohr
 
+try:
+    from chemsh import *
+    from chemsh.io.tools import convert_atoms_to_frag
+
+    chemshell_module = True
+except ImportError:
+    chemshell_module = None
+from monty.dev import requires
 from monty.io import zopen
 from monty.os.path import zpath
 
 if TYPE_CHECKING:
     from numpy.typing import NDArray
 
 
+def get_cluster_info_from_slab(
+    adsorbate_slab_file: Path, slab_center_idx: list[int], adsorbate_idx: list[int]
+) -> tuple[Atoms, Atoms, int, NDArray, NDArray]:
+    """
+    Read the file containing the periodic slab and adsorbate (geometry optimized) and return the key information needed to create an embedded cluster in ChemShell.
+
+    Parameters
+    ----------
+    adsorbate_slab_file
+        The path to the file containing the adsorbate molecule on the surface slab. It can be in any format that ASE can read.
+    adsorbate_idx
+        The indices of the atoms that make up the adsorbate molecule.
+    slab_center_idx
+        The the adsorbate_slab_file, this gives the indices of the atoms that are at the 'center' of the slab right beneath the adsorbate.
+
+    Returns
+    -------
+    adsorbate
```

Comment:
Since the name of the returned variable is not provided, numpy docstring convention suggests using the type. So, `Atoms` instead of `adsorbate`. Same below.